src/Estimators/include/BipedalLocomotion/ContactDetectors/ContactDetector.h

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file ContactDetector.h
+ * @authors Prashanth Ramadoss
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_CONTACT_DETECTOR_H
+#define BIPEDAL_LOCOMOTION_CONTACT_DETECTOR_H
+
+#include <BipedalLocomotion/System/Advanceable.h>
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/Contacts/Contact.h>
+
+#include <iostream>
+#include <unordered_map>
+
+namespace BipedalLocomotion
+{
+namespace Estimators
+{
+
+using ContactStates = std::unordered_map<std::string, BipedalLocomotion::Contacts::EstimatedContact>;
+
+class ContactDetector : public BipedalLocomotion::System::Advanceable<ContactStates>
+{
+public:
+    virtual ~ContactDetector() { };
+
+    /**
+    * Configure generic parameters
+    * @param[in] handler configure the generic parameters for the estimator
+    * @return True in case of success, false otherwise.
+    */
+    bool initialize(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler);
+
+    /**
+    * Compute one step of the detector
+    * The derived class must implement its own methods for setting measurements
+    * and update states to be called within the advance() method
+    * @return True in case of success, false otherwise.
+    */
+    virtual bool advance() final;
+
+    /**
+     * Reset the internal contact states of the detector to false
+     * @return True in case of success, false otherwise.
+     *
+     */
+    virtual bool resetContacts() final;
+
+    /**
+    * Get contact states
+    * @return container of contacts
+    */
+    virtual const ContactStates& get() const final;
+
+    /**
+     * Get state of specific contact
+     * @param[in] contactName name of contact
+     * @return contact state if contact exists, false otherwise
+     */
+    BipedalLocomotion::Contacts::EstimatedContact get(const std::string& contactName);
```

Comment:
Note that the method is not const but it returns by copy. I would suggest returning a const reference and setting the method to ``const``.

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/142/commits/5a0e97011e1545bd422f72a427513f9e5b32d68a.

Cannot just simply return a const reference since it checks whether the contact is existing. So I changed it into a function with an output argument.

Reply:
Ah ok. Sorry, I did not notice that. Then returning by copy makes sense. Simply put the method ``const``.

Reply:
I will have both methods then!

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/142/commits/43c8cf361a8c0fd9d8fe0753ac6e7832ed9841ab